core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java

```diff
@@ -58,14 +60,26 @@ public GcpInstanceTemplateV1Parameters convert(GcpInstanceTemplateV4Parameters s
         return new GcpInstanceTemplateV1Parameters();
     }
 
-    public AzureInstanceTemplateV4Parameters convert(AzureInstanceTemplateV1Parameters source) {
+    public AzureInstanceTemplateV4Parameters convert(AzureInstanceTemplateV1Parameters source, DetailedEnvironmentResponse environment) {
         AzureInstanceTemplateV4Parameters response = new AzureInstanceTemplateV4Parameters();
         response.setEncrypted(source.getEncrypted());
+        initEncryption(response, environment);
         response.setManagedDisk(source.getManagedDisk());
         response.setPrivateId(source.getPrivateId());
         return response;
     }
 
+    private void initEncryption(AzureInstanceTemplateV4Parameters response, DetailedEnvironmentResponse environment) {
+        // FIXME Fetch DES ID from environment; environment can be null!
+        AzureEncryptionV4Parameters encryption = new AzureEncryptionV4Parameters();
+//        encryption.setDiskEncryptionSetId("/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/lrodek-daily-system-rg/providers/" +
```

Comment:
can we remove this?

Reply:
Surely will be removed, once https://jira.cloudera.com/browse/CB-11874 gets resolved that exposes the new field inside `DetailedEnvironmentResponse`.

Reply:
Please note that there are still some FIXME stuffs that will be cleaned up before merging.